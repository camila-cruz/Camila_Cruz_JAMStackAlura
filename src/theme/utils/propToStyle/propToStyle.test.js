import { propToStyle } from '.';

describe('propToStyle()', () => {
  describe('when it receives a single argument', () => {
    describe('it should return a style object', () => {
      test('if the argument is a string', () => {
        const cssProp = 'display';
        const componentProps = { [cssProp]: 'flex' };

        const resultingFunction = propToStyle(cssProp);
        const resultingStyle = resultingFunction(componentProps);

        expect(resultingStyle).toEqual(componentProps);
      });

      test('and if the argument is a number', () => {
        const cssProp = 'flex';
        const componentProps = { [cssProp]: 1 };

        const resultingFunction = propToStyle(cssProp);
        const resultingStyle = resultingFunction(componentProps);

        expect(resultingStyle).toEqual(componentProps);
      });
    });
  });

  describe('when it receives an object with breakpoints', () => {
    describe('it should return a breakpoints object with', () => {
      test('the one specified breakpoint', () => {
        const cssProp = 'color';
        const componentProps = {
          [cssProp]: {
            lg: 'acqua',
          },
        };

        const resultingFunction = propToStyle(cssProp);
        const resultingStyle = resultingFunction(componentProps);

        expect(resultingStyle).toMatchSnapshot();
      });
      test('two or more of the specified breakpoints', () => {
        const cssProp = 'width';
        const componentProps = {
          [cssProp]: {
            xs: '200px',
            md: '600px',
          },
        };

        const resultingFunction = propToStyle(cssProp);
        const resultingStyle = resultingFunction(componentProps);

        expect(resultingStyle).toMatchSnapshot();
      });
    });
  });

  describe('when it receives an invalid argument', () => {
    test('it should throw an error', () => {
      const cssProp = 'margin';
      const componentProps = 'top: 10px';

      const resultingFunction = propToStyle(cssProp);

      expect(() => resultingFunction(componentProps)).toThrowError();
    });
  });
});
