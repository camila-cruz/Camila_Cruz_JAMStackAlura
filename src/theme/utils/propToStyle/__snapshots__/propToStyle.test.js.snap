// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`propToStyle() when it receives an object with breakpoints it should return a breakpoints object with the one specified breakpoint 1`] = `
Array [
  Array [
    "@media screen and (min-width:",
    "992",
    "px){",
    "color: acqua;",
    "}",
  ],
]
`;

exports[`propToStyle() when it receives an object with breakpoints it should return a breakpoints object with two or more of the specified breakpoints 1`] = `
Array [
  Array [
    "@media screen and (min-width:",
    "0",
    "px){",
    "width: 200px;",
    "}",
  ],
  Array [
    "@media screen and (min-width:",
    "768",
    "px){",
    "width: 600px;",
    "}",
  ],
]
`;

exports[`propToStyle() when it receives an object with breakpoints it should return a breakpoints object with two or more of the specified breakpoints again 1`] = `
Array [
  Array [
    "@media screen and (min-width:",
    "0",
    "px){",
    "order: 2;",
    "}",
  ],
  Array [
    "@media screen and (min-width:",
    "768",
    "px){",
    "order: 0;",
    "}",
  ],
]
`;
